mifosng-android/src/main/java/com/mifos/utils/MFErrorParser.java

```diff
@@ -18,8 +21,11 @@
 
     private static Gson gson = new Gson();
 
-    public static MifosError parseError(String serverResponse) {
-        return gson.fromJson(serverResponse, MifosError.class);
+    public static MifosError parseError(String serverResponse) throws JSONException {
+        JSONObject json = new JSONObject(serverResponse);
+        MifosError error = gson.fromJson(serverResponse, MifosError.class);
+        error.setDeveloperMessage((String) json.get("message"));
```

Comment:
I can see, On 400 we are getting this 
`{"timestamp":1534320437227,"status":400,"error":"Bad Request","message":"The tenant identifier: 66 is not valid."}`

Instead of this, Add these fields (message, error, status etc) in the MifosError.

Reply:
Can you please elaborate on this issue @therajanmaurya. I would like to work on this issue